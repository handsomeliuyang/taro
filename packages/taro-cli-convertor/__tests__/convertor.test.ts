'use strict'

import Convertor from '../src/index'
import { clearMockFiles, getResMapFile, setMockFiles } from './__mocks__/fs-extra.js'
import { DEMO_JS_FILE_INFO, DEMO_JS_FILE_INFO_MINIPROGRANROOT, root } from './data/fileData'

describe('微信小程序转换', () => {
  beforeAll(() => {
    // mock报告生成
    jest.spyOn(Convertor.prototype, 'generateReport').mockImplementation(() => {})

    // 配置文件生成
    jest.spyOn(Convertor.prototype, 'generateConfigFiles').mockImplementation(() => {})
  })

  beforeEach(() => {
    // 清空文件信息
    clearMockFiles()
  })

  test('小程序demo转换', () => {
    // 设置初始文件信息
    setMockFiles(root, DEMO_JS_FILE_INFO)

    const convertor = new Convertor(root, false)
    convertor.run()
    const resFileMap = getResMapFile()
    expect(resFileMap).toMatchSnapshot()
  })

  test('配置miniprogramRoot的demo小程序转换', () => {
    // 设置初始文件信息
    setMockFiles(root, DEMO_JS_FILE_INFO_MINIPROGRANROOT)

    const convertor = new Convertor(root, false)
    convertor.run()
    const resFileMap = getResMapFile()
    expect(resFileMap).toMatchSnapshot()
  })
})
