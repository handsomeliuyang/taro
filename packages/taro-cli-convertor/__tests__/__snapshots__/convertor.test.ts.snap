// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`微信小程序转换 小程序demo转换 1`] = `
Map {
  "/wxProject/app.js" => "import withWeapp, { cacheOptions } from \\"@tarojs/with-weapp\\";
import { Block } from \\"@tarojs/components\\";
import React from \\"react\\";
import Taro from \\"@tarojs/taro\\";
cacheOptions.setOptionsToCache({});
@withWeapp(cacheOptions.getOptionsFromCache(), true)
class App extends React.Component {
  render() {
    return this.props.children;
  }
}
export default App;",
  "/wxProject/app.config.js" => "export default { pages: ['pages/index/index'] }
",
  "/wxProject/pages/index/index.js" => "import withWeapp, { cacheOptions } from '@tarojs/with-weapp'
import { Block, View, Text } from '@tarojs/components'
import React from 'react'
import Taro from '@tarojs/taro'
const app = Taro.getApp()
cacheOptions.setOptionsToCache({
  data: {
    motto: 'Hello World',
  },
  onLoad() {},
})
@withWeapp(cacheOptions.getOptionsFromCache())
class _C extends React.Component {
  render() {
    const { motto } = this.data
    return (
      <View>
        <Text>{motto}</Text>
      </View>
    )
  }
}
export default _C
",
  "/wxProject/pages/index/index.config.js" => "export default {}
",
}
`;

exports[`微信小程序转换 配置miniprogramRoot的demo小程序转换 1`] = `
Map {
  "/wxProject/taroConvert/src/app.js" => "import withWeapp, { cacheOptions } from \\"@tarojs/with-weapp\\";
import { Block } from \\"@tarojs/components\\";
import React from \\"react\\";
import Taro from \\"@tarojs/taro\\";
cacheOptions.setOptionsToCache({});
@withWeapp(cacheOptions.getOptionsFromCache(), true)
class App extends React.Component {
  render() {
    return this.props.children;
  }
}
export default App;",
  "/wxProject/taroConvert/src/app.config.js" => "export default { pages: ['pages/index/index'] }
",
  "/wxProject/taroConvert/src/pages/index/index.js" => "import withWeapp, { cacheOptions } from '@tarojs/with-weapp'
import { Block, View, Text } from '@tarojs/components'
import React from 'react'
import Taro from '@tarojs/taro'
const app = Taro.getApp()
cacheOptions.setOptionsToCache({
  data: {
    motto: 'Hello World',
  },
  onLoad() {},
})
@withWeapp(cacheOptions.getOptionsFromCache())
class _C extends React.Component {
  render() {
    const { motto } = this.data
    return (
      <View>
        <Text>{motto}</Text>
      </View>
    )
  }
}
export default _C
",
  "/wxProject/taroConvert/src/pages/index/index.config.js" => "export default {}
",
}
`;
